### PR TITLE
Add config to integrate with circle ci

### PR DESCRIPTION
**Pull request by Pradeepkumarbk**
-  Added a circle.yaml for integrating Scope with Circle CI.
-  Whenever a Commit  is made or PR is raised the CI triggers  and builds a docker image .
- Pushes the docker image onto dockerhub when PR is merged respectively i.e.
        If PR is merged against `staging`branch - image with `tag` will be pushed to docker hub repo.
        If PR is merged against `master`branch - image with `ci` will be pushed to docker hub repo.

Signed-off-by: Pradeepkumarbk <pradeepkumar.bk96@gmail.com>

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -32,86 +32,39 @@ dependencies:
...
-#      - go get github.com/weaveworks/wordepress && cd /home/ubuntu/src/github.com/weaveworks/wordepress && git checkout v1.0.0 && cd cmd/wordepress && go get
-#      - cd $SRCDIR; PRODUCT=scope tools/publish-site "$WP_LIVE_URL" "$WP_LIVE_USER" "$WP_LIVE_PASSWORD"
-#  issues:
-#    branch: /.*/
-#    owner: weaveworks
-#    commands:
-#      - go get github.com/weaveworks/wordepress && cd /home/ubuntu/src/github.com/weaveworks/wordepress && git checkout v1.0.0 && cd cmd/wordepress && go get
-#      - cd $SRCDIR; PRODUCT=scope tools/publish-site "$WP_DEV_URL" "$WP_DEV_USER" "$WP_DEV_PASSWORD"
+      - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
+      - docker push openebs/scope:latest
```
Comment (satyamz) [inline circle.yml]: Please do not push these to the `openebs/scope:lastest` change it to `openebs/scope:ci`

File: docker/Dockerfile.scope [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM weaveworks/cloud-agent
+FROM openebs/cloud-agent
```
Comment (satyamz) [inline docker/Dockerfile.scope]: I think we also don't need this change. Instead let it build and tag the final image with openebs prefix

File: extras/Makefile [hunk 3]
```diff
@@ -10,7 +10,7 @@ all: $(TARGETS)
 ifeq ($(BUILD_IN_CONTAINER),true)
 $(TARGETS):
 	$(SUDO) docker run -ti $(RM) -v $(shell pwd)/../:/go/src/github.com/weaveworks/scope -e GOARCH -e GOOS \
-		weaveworks/scope-backend-build -C extras $@
+		openebs/scope-backend-build -C extras $@
```
Comment (satyamz) [inline extras/Makefile]: same comment apply here too

**Review by satyamz — CHANGES_REQUESTED** [review]

File: circle.yml [hunk 4]
```diff
@@ -32,86 +32,39 @@ dependencies:
...
-    - "cd $SRCDIR/backend; ../tools/rebuild-image weaveworks/scope-backend-build . Dockerfile build.sh && touch $SRCDIR/.scope_backend_build.uptodate"
+    - "cd $SRCDIR/client; ../tools/rebuild-image openebs/scope-ui-build . Dockerfile package.json webpack.production.config.js .eslintrc .babelrc && touch $SRCDIR/.scope_ui_build.uptodate"
+    - "cd $SRCDIR/backend; ../tools/rebuild-image openebs/scope-backend-build . Dockerfile build.sh && touch $SRCDIR/.scope_backend_build.uptodate"
     - test -z "$SECRET_PASSWORD" || (cd $SRCDIR/integration; ./gce.sh make_template):
         parallel: false
     - sudo apt-get update && sudo apt-get install python-pip && sudo pip install awscli
 
 test:
   override:
-    - cd $SRCDIR; make RM= lint:
```
Comment (qiell) [inline circle.yml]: why is this removed? I think we should check linting errors before creating a build.

File: circle.yml [hunk 5]
```diff
@@ -32,86 +32,39 @@ dependencies:
...
-        parallel: true
     - cd $SRCDIR; COVERDIR=./coverage make RM= CODECGEN_UID=23 tests:
         parallel: true
     - cd $SRCDIR; make RM= client-test static:
         parallel: true
-    - cd $SRCDIR; make RM= client-lint static:
```
Comment (qiell) [inline circle.yml]: same here.

**Review by qiell — CHANGES_REQUESTED** [review]
@Pradeepkumarbk I agree with @satyamz 
Let's build the image using weaveworks and later we can tag it and push it.

**Review by satyamz — APPROVED** [review]
lgtm @Pradeepkumarbk Can you change the commit message? Its not good.

**Review by qiell — APPROVED** [review]
lgtm

**Pull request merged**